test/particle.test.ts

```diff
@@ -0,0 +1,62 @@
+import { Random, RngStateBuiltin, RngStateObserved } from "@reside-ic/random";
+
+import { base } from "../src/base";
+import { Particle } from "../src/particle";
+
+import * as models from "./models";
+
+describe("Can create a particle", () => {
+    const pars = {n: 1, sd: 1};
+    it("is created", () => {
+        const r = new RngStateObserved(new RngStateBuiltin());
+        const s = new Random(r);
+        const cmp = new Random(r.replay());
+
+        const p = new Particle(new models.Walk(base, pars), 0);
+        p.run(2, s);
+        expect(p.state(null))
+            .toEqual([cmp.randomNormal() + cmp.randomNormal()]);
+    });
+
+    it("Can return state", () => {
+        const p = new Particle(new models.Walk(base, pars), 0);
+        expect(p.size).toEqual(1);
+        expect(p.state(null)).toEqual([0]);
+        expect(p.state([0])).toEqual([0]);
+        expect(p.state([0, 0])).toEqual([0, 0]);
```

Comment:
so this is legit even though the size of the model here is just 1? or is it just that validation is coming in a future branch?

Reply:
this is legit (though it would be unusual) in dust too. However, this is all gone in 6dffcb8